pkgs/applications/science/electronics/vivado/default.nix

```diff
@@ -0,0 +1,127 @@
+  pname = "vivado";
+  version = "2017.2";
+
+  buildInputs = [
+    patchelf
+    procps
+    ncurses5
+    makeWrapper
+    coreutils
+  ];
+
+  inherit ncurses5;
+
+  # requireFile prevents rehashing each time, which saves time during
+  # rebuilds.
+  src = requireFile rec {
+    name = "Xilinx_Vivado_SDK_2017.2_0616_1.tar.gz";
+    message = ''
+      This nix expression requires that ${name} is already part of the store.
+      Login to Xilinx, download from
+      https://www.xilinx.com/support/download.html,
+      rename the file to ${name}, and add it to the nix store with
+      "nix-prefetch-url file:///path/to/${name}".
+    '';
+    sha256 = "06pb4wjz76wlwhhzky9vkyi4aq6775k63c2kw3j9prqdipxqzf9j";
+  };
+
+  patchPhase = ''
+    patchShebangs .
+    patchelf --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" \
+         tps/lnx64/jre/bin/java
+  '';
+
+  dontBuild = true;
+
+  installPhase = ''
+    # generate an installation config file
+    cat <<EOF > install_config.txt
+    Edition=Vivado HL WebPACK
+    Destination=$out/opt
+    Modules=Software Development Kit (SDK):1,DocNav:0,Kintex UltraScale:0,Zynq-7000:1,System Generator for DSP:0,Artix-7:1,Kintex-7:0
+    InstallOptions=Acquire or Manage a License Key:0,Enable WebTalk for SDK to send usage statistics to Xilinx:0
+    CreateProgramGroupShortcuts=0
+    ProgramGroupFolder=Xilinx Design Tools
+    CreateShortcutsForAllUsers=0
+    CreateDesktopShortcuts=0
+    CreateFileAssociation=0
+    EOF
+
+    ($(pwd)/xsetup --agree 3rdPartyEULA,WebTalkTerms,XilinxEULA --batch Install --config install_config.txt || true) | while read line
+    do
+        [[ "''${line}" == *"Execution of Pre/Post Installation Tasks Failed"* ]] && echo "killing installer!" && ((pkill -9 -f "tps/lnx64/jre/bin/java") || true)
+        echo ''${line}
+    done
+
+    # Add vivado and xsdk to bin folder
+    mkdir $out/bin
+    ln -s $out/opt/Vivado/2017.2/bin/vivado $out/bin/vivado
+    ln -s $out/opt/SDK/2017.2/bin/xsdk $out/bin/xsdk
+  '';
+
+  postFixup = ''
+    # Hack around lack of libtinfo in NixOS
+    ln -s $ncurses/lib/libncursesw.so.6 $out/opt/Vivado/2017.2/lib/lnx64.o/libtinfo.so.5
+    ln -s $ncurses/lib/libncursesw.so.6 $out/opt/SDK/2017.2/lib/lnx64.o/libtinfo.so.5
+
+    wrapProgram $out/opt/Vivado/2017.2/bin/vivado --prefix LD_LIBRARY_PATH : "$libPath"
+    wrapProgram $out/opt/SDK/2017.2/bin/xsdk --prefix LD_LIBRARY_PATH : "$libPath"
+    wrapProgram $out/opt/SDK/2017.2/eclipse/lnx64.o/eclipse --prefix LD_LIBRARY_PATH : "$libPath"
+    wrapProgram $out/opt/SDK/2017.2/tps/lnx64/jre/bin/java --prefix LD_LIBRARY_PATH : "$libPath"
+
+    # wrapProgram on its own will not work because of the way the Vivado script runs ./launch
+    # Therefore, we need Even More Patches...
+    sed -i -- 's|`basename "\$0"`|vivado|g' $out/opt/Vivado/2017.2/bin/.vivado-wrapped
+    sed -i -- 's|`basename "\$0"`|xsdk|g' $out/opt/SDK/2017.2/bin/.xsdk-wrapped
+  '';
+
+  libPath = lib.makeLibraryPath [
```

Comment:
Please move this into a let in to postFixup

Reply:
I get an error when I try to run vivado after doing this: `application-specific initialization failed: couldn't load file "librdi_commontasks.so": libtinfo.so.5: cannot open shared object file: No such file or directory`. It works if I keep it out of the let though.

Reply:
Yes, this is because your are using bash variables instead of nix ones in preFixup. Please use nix variables there.